### ISSUE TITLE
UnnecessaryAbstractClass excludeAnnotatedClasses not working

### ISSUE THREAD

**Issue by jaychang0917** [arturbosch/detekt#2360]
## Expected Behavior
The follow code should pass the rule `UnnecessaryAbstractClass`.
```kotlin
package com.jaychang.test

import dagger.Binds

@dagger.Module
abstract class Module {
    @Binds
    abstract fun provideFoo(foo: Foo): Foo
}
```

The config
```
UnnecessaryAbstractClass:
    active: true
    excludeAnnotatedClasses: "dagger.Module"
```

## Observed Behavior
It fails the rule.

## Your Environment

* Version of detekt used: 1.5.1
* Version of Gradle used (if applicable): 5.4.1
* Operating System and version: Mac OS

Thanks!

**Comment by cortinico** [comment]
I was able to reproduce this issue. I'm working on a potential fix

**Comment by cortinico** [comment]
Fix is here: #2361

Apparently we're not catching fully qualified annotations (like yours `@dagger.Module`).
Just a couple of side note to your issue:

1. Specifying `excludeAnnotatedClasses: "dagger.Module"` is superfluous. That's the default value for that property: https://arturbosch.github.io/detekt/style.html#unnecessaryabstractclass

2. Your code does not compile as you're missing the `import dagger.Module` (I assume that was a mistake).

**Comment by ggv-jaychang** [comment]
Thanks!

> 2. Your code does not compile as you're missing the `import dagger.Module` (I assume that was a mistake).

Actually the code compiles.

**Comment by jaychang0917** [comment]
Thanks.

> 2. Your code does not compile as you're missing the `import dagger.Module` (I assume that was a mistake).

Actually the code compiles.

**Comment by cortinico** [comment]
> Actually the code compiles.

Yeah you're right, my fault.
Unfortunately this rule relies on imports to work so I had to rework it a bit.
